public/components/common/modules/panel/components/vis-card.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Wazuh app - React component VisCard.
+ *
+ * Copyright (C) 2015-2021 Wazuh, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Find more information about this on the LICENSE file.
+ */
+
+import React, { useState } from 'react';
+import { EuiButtonIcon, EuiFlexGroup, EuiFlexItem, EuiPanel, EuiTitle } from '@elastic/eui';
+import { RawVisualizations } from '../../../../../factories/raw-visualizations';
+import KibanaVis from '../../../../../kibana-integrations/kibana-vis';
+
+export const VisCard = ({ changeView = () => {}, id, width, tab, ...props }) => {
+  const [expandedVis, setExpandedVis] = useState(false);
+
+  const title = (() => {
+    const visList = new RawVisualizations().getList();
+    const rawVis = visList ? visList.filter((item) => item && item.id === id) : [];
+    return rawVis.length && rawVis[0]?.attributes?.title;
+  })();
+
+  const toggleExpand = (id) => {
+    setExpandedVis(expandedVis === id ? false : id);
+  };
+
+  return (
+    <>
+      {' '}
+      <EuiFlexItem grow={width}>
+        <EuiPanel paddingSize={'s'} className={expandedVis === id ? 'fullscreen h-100' : 'h-100'}>
```

Comment:
Since this is a singular component, and it's not handling whether other `vis-card` are expanded or not, why check for equality instead of using a boolean state.